lib/css_parser/parser.rb

```diff
@@ -185,6 +185,27 @@ def each_rule_set(media_types = :all) # :yields: rule_set, media_types
       end
     end
 
+    # Output all CSS rules as a Hash
+    def to_h(media_types = :all)
+      out = {}
+       styles_by_media_types = {}
+       each_selector(media_types) do |selectors, declarations, specificity, media_types|
+         media_types.each do |media_type|
+           styles_by_media_types[media_type] ||= []
+           styles_by_media_types[media_type] << [selectors, declarations]
+         end
+       end
+
+      styles_by_media_types.each_pair do |media_type, media_styles|
+        ms = {}
+        media_styles.each do |media_style|
```

Comment:
might be able to do this in a single loop by doing `ms[media_type] ||= {}` and then using this as what is now `ms`